libs/pageserver_api/src/models.rs

```diff
@@ -149,30 +151,86 @@ impl std::ops::Deref for TenantCreateRequest {
     }
 }
 
+pub fn deserialize_option_number_from_string<'de, T, D>(
+    deserializer: D,
+) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: FromStr + serde::Deserialize<'de>,
+    <T as FromStr>::Err: Display,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum NumericOrNull<'a, T> {
+        String(String),
+        Str(&'a str),
```

Comment:
At least this `Str` variant is too much, or do you have an example where it ends up being used? I think it would require `#[serde(borrow)]` at least.

Might be more clear to just do all this with a plain visitor impl and `deserializer.deserialize_any`. One thing that will become an issue is the `usize`/`u64` handling. I wonder if there was a way to piggy-back on Deserializer to have `T` deserialize itself from the type, or from within a string, that would solve the signedness, bounds, and remove the need to use `FromStr`.